hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/contracts/execution/HederaEvmMessageCallProcessorV045.java

```diff
@@ -0,0 +1,158 @@
+                    return;
+                }
+            } else if (frameHasValue) {
+                final var updater = (AbstractLedgerEvmWorldUpdater) frame.getWorldUpdater();
+                if (updater.isTokenAddress(frame.getRecipientAddress())) {
+                    frame.setExceptionalHaltReason(ILLEGAL_STATE_CHANGE);
+                    frame.setState(MessageFrame.State.EXCEPTIONAL_HALT);
+                    operationTracer.tracePostExecution(
+                            frame,
+                            new Operation.OperationResult(
+                                    frame.getRemainingGas(), ExceptionalHaltReason.ILLEGAL_STATE_CHANGE));
+                    return;
+                } else if (updater.get(frame.getRecipientAddress()) == null) {
+                    executeLazyCreate(frame, operationTracer);
+                    if (frame.getState() == EXCEPTIONAL_HALT) {
+                        return;
+                    }
+                }
+            }
+
+            ((AbstractLedgerEvmWorldUpdater) frame.getWorldUpdater())
+                    .setCreationCustomizerForSponsor(frame.getSenderAddress());
+
+            super.start(frame, operationTracer);
+        }
+    }
+
+    protected void executeHederaPrecompile(
+            final PrecompiledContract contract, final MessageFrame frame, final OperationTracer operationTracer) {
+        if (contract instanceof EvmHTSPrecompiledContract htsPrecompile) {
+            var updater = (AbstractLedgerEvmWorldUpdater) frame.getWorldUpdater();
+            final var costedResult = htsPrecompile.computeCosted(
+                    frame.getInputData(),
+                    frame,
+                    (now, minimumTinybarCost) -> minimumTinybarCost,
+                    updater.tokenAccessor());
+            output = costedResult.getValue();
+            gasRequirement = costedResult.getKey();
+        }
+        if (!"HTS".equals(contract.getName()) && !"EvmHTS".equals(contract.getName())) {
+            output = contract.computePrecompile(frame.getInputData(), frame).getOutput();
+            gasRequirement = contract.gasRequirement(frame.getInputData());
+        }
+
+        operationTracer.tracePrecompileCall(frame, gasRequirement, output);
+        if (frame.getState() == REVERT) {
+            return;
+        }
+        if (frame.getRemainingGas() < gasRequirement) {
+            frame.decrementRemainingGas(frame.getRemainingGas());
+            frame.setExceptionalHaltReason(Optional.of(INSUFFICIENT_GAS));
+            frame.setState(EXCEPTIONAL_HALT);
+        } else if (output != null) {
+            frame.decrementRemainingGas(gasRequirement);
+            frame.setOutputData(output);
+            frame.setState(COMPLETED_SUCCESS);
+        } else {
+            frame.setState(EXCEPTIONAL_HALT);
+        }
+    }
+
+    protected void executeLazyCreate(final MessageFrame frame, final OperationTracer operationTracer) {
+        // no-op
+    }
+}
```

Comment:
As far as I can tell, there are no differences between this file and the V38 version.  Is this true?  Also the same question for the V45 versions of the operations.  I understand why we need to create these files for the new evm version but for easier review, it would be nice to know if anything has changed in these files.

Reply:
The key difference is [here](https://github.com/hashgraph/hedera-services/pull/9628/files/2f1b95f71a58298c7a37b620176aff1d76f7e2fe#diff-f6b7d8a2463cb67f670dc2693b00f0e1223a6bef6e0271e4ba0a4ef5e7f30bb4R114). And most of the v45 files are same as v38 but using [HederaOperationUtilV045](https://github.com/hashgraph/hedera-services/pull/9628/files/2f1b95f71a58298c7a37b620176aff1d76f7e2fe#diff-c7199b2acd7fd4402b6310305c8b7f9ee1fdbc94551a2a853561b722a2cb1e82R42) where the `INVALID_SOLIDITY_ADDRESS` check had been removed. I would say that the main changes has been done here:
- https://github.com/hashgraph/hedera-services/pull/9628/files#diff-f6b7d8a2463cb67f670dc2693b00f0e1223a6bef6e0271e4ba0a4ef5e7f30bb4R114
- https://github.com/hashgraph/hedera-services/pull/9628/files#diff-e10c913db9f6cd627bf87ae17a80510dca59e8792eb0ac2b59d0ef967604b658R134
- https://github.com/hashgraph/hedera-services/pull/9628/files#diff-fdd7b93ba105216deef18b34063222d969680bfd5a50cab9b7726a34705358f5R91
- https://github.com/hashgraph/hedera-services/pull/9628/files#diff-276868df7723ffbcb87082bf04260f1ad6925821098a6e4f1541f9dc7e5b3a49R177
- https://github.com/hashgraph/hedera-services/pull/9628/files#diff-c7199b2acd7fd4402b6310305c8b7f9ee1fdbc94551a2a853561b722a2cb1e82R117
- https://github.com/hashgraph/hedera-services/pull/9628/files#diff-096e95e56d8349fd27a7a2c7ff113184b9fab4de0459a5c365cb790a866aea8cR73
- https://github.com/hashgraph/hedera-services/pull/9628/files#diff-f4a9c9a79bad422339d5033268ac44c97c68d2971863c8aedaf3e855bc6f3c76R80
- https://github.com/hashgraph/hedera-services/pull/9628/files#diff-bef7a524d6ec9a8a34a10f0965b760c92a2c4ed87dddbb4db2580a843dcdc7d0R140

Reply:
I suggest adding these key code changes as a list in the PR description with a small description of what each change is for, that will make reviewing this PR a lot easier